calyx-py/calyx/builder.py

```diff
@@ -268,53 +277,64 @@ def seq_mem_d1(
             name, ast.Stdlib.seq_mem_d1(bitwidth, len, idx_size), is_external, is_ref
         )
 
-    def add(self, name: str, size: int, signed=False) -> CellBuilder:
-        """Generate a StdAdd cell."""
+    def binary_op_helper(
```

Comment:
While I know this was introduced as just a way to cut down on boilerplate, I think we should consider "publishing" it as an actual public API that generators should use. That would mean picking a name that goes along with stuff like `lt` and `sub` or whatever, i.e., kinda looking like it's "declaring" the thing instead of calling a function to generate it. (That's what's makes this aesthetically an eDSL rather than "just" a library.)

I'd recommend `binary` but would also go along with `binary_op`.